### PR TITLE
Log a warning when --nocache is passed

### PR DESCRIPTION
**Pull request by peterebden**
First step of #1212

### CODE REVIEW

File: src/please.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ var opts struct {
 
 	FeatureFlags struct {
```
Comment (Tatskaari) [inline src/please.go:73]: This is confusingly named. Feature flags are what I'm calling the things that gate breaking changes between releases. Might want to revisit this.

**Review by Tatskaari — APPROVED** [review]

File: src/please.go (hunk 1 above)
Comment (peterebden) [reply src/please.go:73]: This was there first :-P

But agreed. It's only internal to this package so we can call it whatever we want.

Comment (Tatskaari) [reply src/please.go:73]: Oh okay, I assumed this affected `plz halp`

Comment (peterebden) [reply src/please.go:73]: This is the struct that defines the "Options that enable / disable certain features" section in `plz --help`, but the actual name of the struct doesn't appear there.

**Pull request merged**